app/models/manageiq/providers/ibm_power_hmc/infra_manager/event_target_parser.rb

```diff
@@ -70,14 +68,36 @@ def handle_usertask(usertask)
     if usertask["status"].eql?("Completed")
       case usertask["key"]
       when "TEMPLATE_PARTITION_SAVE", "TEMPLATE_PARTITION_SAVE_AS", "TEMPLATE_PARTITION_CAPTURE"
-        r = {:assoc => :miq_templates, :ems_ref => usertask['template_uuid']}
+        handle_usertask_template_save(usertask)
       when "TEMPLATE_DELETE"
-        template = ManageIQ::Providers::InfraManager::Template.find_by(:ems_id => ems_event.ext_management_system.id, :name => usertask['labelParams'])
-        unless template.nil?
-          r = {:assoc => :miq_templates, :ems_ref => template.uid_ems}
-        end
+        handle_usertask_template_delete(usertask)
+      when "PCM_PREFERENCE_UPDATE"
+        handle_usertask_pcm_preference(usertask)
+      else
+        []
+      end
+    end
+  end
+
+  def handle_usertask_template_save(usertask)
+    [{:assoc => :miq_templates, :ems_ref => usertask['template_uuid']}]
+  end
+
+  def handle_usertask_template_delete(usertask)
+    template = ManageIQ::Providers::InfraManager::Template.find_by(:ems_id => ems_event.ext_management_system.id, :name => usertask['labelParams'])
+    if template.nil?
+      []
+    else
+      [{:assoc => :miq_templates, :ems_ref => template.uid_ems}]
+    end
+  end
+
+  def handle_usertask_pcm_preference(usertask)
+    usertask['labelParams'].first.tr("[] ", "").split(",").each_with_object([]) do |hostname, array|
+      host = ManageIQ::Providers::IbmPowerHmc::InfraManager::Host.find_by(:ems_id => ems_event.ext_management_system.id, :name => hostname)
```

Comment:
Minor but it is better to use the association rather than the direct IDs (if the assoc is cached it can use that and not do a query)

```suggestion
      host = ManageIQ::Providers::IbmPowerHmc::InfraManager::Host.find_by(:ext_management_system => ems_event.ext_management_system, :name => hostname)
```